source/draw/gpu/GpuImageBin.ooc

```diff
@@ -14,63 +14,64 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
-
+use ooc-collections
 use ooc-math
 use ooc-base
-import structs/FreeArrayList, GpuImage, GpuMonochrome, GpuBgr, GpuBgra, GpuUv, GpuYuv420Semiplanar, GpuYuv420Planar, GpuYuv422Semipacked
+import GpuImage, GpuMonochrome, GpuBgr, GpuBgra, GpuUv, GpuYuv420Semiplanar, GpuYuv420Planar, GpuYuv422Semipacked
 import threading/Thread
 
 GpuImageBin: class {
-	_monochrome: FreeArrayList<GpuImage>
-	_bgr: FreeArrayList<GpuImage>
-	_bgra: FreeArrayList<GpuImage>
-	_uv: FreeArrayList<GpuImage>
-	_yuv422: FreeArrayList<GpuImage>
+	_monochrome: VectorList<GpuImage>
+	_bgr: VectorList<GpuImage>
+	_bgra: VectorList<GpuImage>
+	_uv: VectorList<GpuImage>
+	_yuv422: VectorList<GpuImage>
 	_mutex: Mutex
 	_limit := 15
 
 	init: func {
 		this _mutex = Mutex new()
-		this _monochrome = FreeArrayList<GpuImage> new()
-		this _bgr = FreeArrayList<GpuImage> new()
-		this _bgra = FreeArrayList<GpuImage> new()
-		this _uv = FreeArrayList<GpuImage> new()
-		this _yuv422 = FreeArrayList<GpuImage> new()
+		this _monochrome = VectorList<GpuImage> new()
+		this _bgr = VectorList<GpuImage> new()
+		this _bgra = VectorList<GpuImage> new()
+		this _uv = VectorList<GpuImage> new()
+		this _yuv422 = VectorList<GpuImage> new()
+	}
+	_cleanList: func (list: VectorList<GpuImage>) {
+		list apply(|image| image _recyclable = false)
```

Comment:
Looks to me like this creates a Func, which you then have no way of freeing afterwards.